### PR TITLE
Add inplace flag sort to sort operation

### PR DESCRIPTION
**Pull request by ducleminh**
### Description and related tickets, documents
Add inplace flag sort to sort operation

Reviewers: @hai-adatao @nhanitvn @Huandao0812 @phvu 

### Breaking changes & backward compatible issues
Is this PR a breaking change or has backward compatible issue (e.g: changes in API names, interfaces, signature / remove something...)?
If yes, please state out what and tag, and if it can possibly affect BA, please tag aht, tuananh, khangpham and baonguyen

### How to test
Describe how this PR is tested. In case manual testing is required, describe how to do so.

### PR Progress
Make sure all checkboxes below are checked before merged
- [x] Branch is in format `prefix/description` (see [this](http://www.guyroutledge.co.uk/blog/git-branch-naming-conventions/))
- [x] Merge check has no conflicts. PR checks passed.
- [x] Code review is done.

### CODE REVIEW

File: core/src/main/java/io/ddf/etl/TransformationHandler.java [hunk 1]
```diff
@@ -400,6 +400,16 @@ public DDF sort(List<String> columns, List<Boolean> ascending) throws DDFExcepti
     return newDDF;
   }
 
+  @Override
+  public DDF sort(List<String> columns, List<Boolean> ascending, Boolean inPlace) throws DDFException {
```
Comment (phvu) [inline core/src/main/java/io/ddf/etl/TransformationHandler.java]: Similar comment, maybe move the real implementation into this function?

File: ddf-test/src/main/scala/io/ddf/test/it/etl/TransformationHandlerBaseSuite.scala [hunk 2]
```diff
@@ -204,4 +204,28 @@ trait TransformationHandlerBaseSuite extends BaseSuite with Matchers {
       smithsData.Transform.sort(columns, new util.ArrayList[Boolean]())
     }
   }
+
+  test("test sort inPlace") {
```
Comment (hai-adatao) [inline ddf-test/src/main/scala/io/ddf/test/it/etl/TransformationHandlerBaseSuite.scala]: Please make 2 different tests instead

File: core/src/main/java/io/ddf/etl/TransformationHandler.java (hunk 1 above)
Comment (ducleminh) [inline core/src/main/java/io/ddf/etl/TransformationHandler.java]: refactored

File: ddf-test/src/main/scala/io/ddf/test/it/etl/TransformationHandlerBaseSuite.scala (hunk 2 above)
Comment (ducleminh) [inline ddf-test/src/main/scala/io/ddf/test/it/etl/TransformationHandlerBaseSuite.scala]: refactored

File: ddf-test/src/main/scala/io/ddf/test/it/etl/TransformationHandlerBaseSuite.scala [hunk 3]
```diff
@@ -204,4 +204,38 @@ trait TransformationHandlerBaseSuite extends BaseSuite with Matchers {
       smithsData.Transform.sort(columns, new util.ArrayList[Boolean]())
     }
   }
+
+  test("test sort inPlace false") {
+    val ddf: DDF = loadSmithsDDF().VIEWS.project("subject, variable, value")
+    val columns = new util.ArrayList[String]()
```
Comment (nhanitvn) [inline ddf-test/src/main/scala/io/ddf/test/it/etl/TransformationHandlerBaseSuite.scala]: Is there a shorter way to specify columns and ascending? May be construct a Scala list and convert to the Java one

Comment (nhanitvn) [inline ddf-test/src/main/scala/io/ddf/test/it/etl/TransformationHandlerBaseSuite.scala]: It is OK to merge, though, :)

**Pull request merged**